examples/meta/src/evaluation/cross_validation_pipeline.sg

```diff
@@ -14,7 +14,7 @@ Labels labels_test = labels(f_labels_test)
 Transformer subMean = transformer("PruneVarSubMean")
 Machine svm = machine("LibLinear")
 
-PipelineBuilder builder()
+PipelineBuilder builder = pipeline()
```

Comment:
exactly like this! :)

Reply:
BTW why is the C++ ype not just called Pipeline and the Machine called PipelineMachine?

Reply:
i think `Pipeline` and `PipelineMachine` might be a bit confusing, while `PipelineBuilder` can indicate its usage as a builder

Reply:
TBH, I dont really agree, I think `then` should actually return `CMachine` (since the subsequent object will be used in this fashion) ... why do we need to know that a machine is a pipeline if it behaves as a machine?
And then `Pipeline` is the thing that builds the stuff.
Makes for a cleaner API imo

@vigsterkr @lisitsyn @iglesias what are your thoughts?

Reply:
@karlnapf are we talking now about PipelineMachine and PipelineBuilder, or why `then` returns Pipeline* ? :) imo those are different things or?

Reply:
in case `then` would return CMachine, then the whole thing becomes opaque, and you would never be able to get out the elements of pipeline...
unless we could register `std::vector<std::pair<std::string, variant<CTransformer*, CMachine*>>>  m_stages;` and then use getter... which is currently a nogo for multiple reasons :)

Reply:
Sorry I wasnt clear ;)

What I suggested is this: CPipeline inherits from CMachine, so it *is* and behaves as a machine. Therefore I don't see a need to expose a separate class (say to swig). All we do is to call train/etc. I would call this thing CPipelineMachine and hide it from the outside world.

Next, the class that constructs the above machine (whose name is now hidden from the API) could be called CPipeline now, the thing where one can add stages etc and the finalise as a CMachine.

Does it make any sense?

Reply:
So CPipeline -> CPipelineMachine, CPipelineBuilder ->CPipeline, then returns CMachine

Reply:
imo builder is just clearer but i dont have any strong feelings about it...
note my second comment about extraction and observation of pipeline.

Reply:
yes the stages thing is a problem

Reply:
i.e. until we cannot register and cleanly extract stages from a pipeline this sort of explicit exposure is required :) otherwise the whole thing becomes totally opaque once built

Reply:
got it! thx for clarifying.
I still would slightly prefer different names (CPipelineMachine, CPipeline) but it is only a minor difference, also no strong feelings. Can leave it as it is